src/sentry/preprod/api/endpoints/preprod_artifact_rerun_analysis.py

```diff
@@ -0,0 +1,256 @@
+
+from sentry import analytics
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, region_silo_endpoint
+from sentry.api.permissions import StaffPermission
+from sentry.models.files.file import File
+from sentry.preprod.analytics import PreprodArtifactApiRerunAnalysisEvent
+from sentry.preprod.api.bases.preprod_artifact_endpoint import PreprodArtifactEndpoint
+from sentry.preprod.models import (
+    PreprodArtifact,
+    PreprodArtifactSizeComparison,
+    PreprodArtifactSizeMetrics,
+)
+from sentry.preprod.producer import produce_preprod_artifact_to_kafka
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+class PreprodArtifactRerunAnalysisEndpoint(PreprodArtifactEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+
+    def post(self, request: Request, project, head_artifact_id, head_artifact) -> Response:
+        """
+        User facing endpoint to rerun analysis for a specific preprod artifact.
+        """
+
+        analytics.record(
+            PreprodArtifactApiRerunAnalysisEvent(
+                organization_id=project.organization_id,
+                project_id=project.id,
+                user_id=request.user.id,
+                artifact_id=head_artifact_id,
+            )
+        )
+
+        cleanup_old_metrics(head_artifact)
+        reset_artifact_data(head_artifact)
+
+        try:
+            produce_preprod_artifact_to_kafka(
+                project_id=head_artifact.project.id,
+                organization_id=head_artifact.project.organization_id,
+                artifact_id=head_artifact_id,
+            )
+        except Exception as e:
+            logger.exception(
+                "preprod_artifact.rerun_analysis.kafka_error",
+                extra={
+                    "artifact_id": head_artifact_id,
+                    "user_id": request.user.id,
+                    "organization_id": head_artifact.project.organization_id,
+                    "project_id": head_artifact.project.id,
+                    "error": str(e),
+                },
+            )
+            return Response(
+                {
+                    "error": f"Failed to queue analysis for artifact {head_artifact_id}",
+                },
+                status=500,
+            )
+
+        logger.info(
+            "preprod_artifact.rerun_analysis",
+            extra={
+                "artifact_id": head_artifact_id,
+                "user_id": request.user.id,
+                "organization_id": head_artifact.project.organization_id,
+                "project_id": head_artifact.project.id,
+            },
+        )
+
+        return success_response(
+            artifact_id=head_artifact_id,
+            state=head_artifact.state,
+        )
+
+
+@region_silo_endpoint
+class PreprodArtifactAdminRerunAnalysisEndpoint(Endpoint):
```

Comment:
I decided to put these into the same file, but fine to keep them separate if you prefer, just lmk